test/unit/lib/sqs.spec.js

```diff
@@ -221,7 +222,9 @@ describe('SQS Utilities', function() {
           expect(res).to.deep.equal({ res: result, extended: false });
           expect(sqsMock.sendMessage.callCount).to.equal(1);
           expect(sqsMock.sendMessage.args[0][0]).to.deep.equal({
-            MessageBody: compress(JSON.stringify(payload)).toString('base64'),
+            MessageBody: compress(JSON.stringify(payload), {
+              level: zlib.constants.Z_NO_COMPRESSION
+            }).toString('base64'),
             QueueUrl: queueUrl,
```

Comment:
General question - so if under 64k we are not going to compress.  Should an attribute be set so that the poster doesn't call decompression algorithm on this?  Thinking there may be some overhead there even if the payload is not compressed - unless it can detect that from the payload content without transverse the entire payload.

Reply:
My understanding is that the compression alg will put metadata on the payload, such as the compression level, so that the decompression knows that it doesn't need to attempt decompression, just remove the metadata.